Assets/Scripts/Dialog/DialogSpeaker.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Articy.Harmonybarktest;
+using Articy.Unity;
+using UnityEngine;
+
+
+namespace HarmonyQuest.Dialog
+{
+    [RequireComponent(typeof(SphereCollider))]
+    public class DialogSpeaker : MonoBehaviour
+    {
+        public ArticyRef character;
+        public ArticyRef DialogReference;
+        public GameObject DialogPrefab;
+
+        [HideInInspector]
+        public DialogView dialogView;
+
+        private SphereCollider teaseCollider;
+
+        public static float DialogRadius = 1.5f;
+
+        Camera mainCamera;
+
+        // Start is called before the first frame update
+        void Start()
+        {
+            ServiceLocator.instance.GetDialogController().RegisterSpeaker(this);
+            mainCamera = ServiceLocator.instance.GetCamera();
+
+            GameObject go = Instantiate(DialogPrefab, gameObject.transform);
+
+            dialogView = go.GetComponent<DialogView>();
+            dialogView.Init();
+
+            dialogView.SetTargetForTrackers(transform);
+            if (DialogReference.HasReference)
+            {
+                dialogView.barkText.text = DialogReference.GetObject<HQ_NPC_DIALOGUE>().GetFeatureHQ_NPC().TeaseText;
```

Comment:
We'll want some sort of way to update the bark text, in the case that the player changes a dialogue variable that causes a branch in the bark text without reloading the scene.

i.e. After you return an NPC's special rock he shouldn't still ask where his special rock is in his bark.

Reply:
To clarify, my issue is that this text only updates once in the Start function